bundle/src/main/java/com/adobe/acs/commons/mcp/util/Spreadsheet.java

```diff
@@ -191,23 +204,135 @@ public int getRowCount() {
 
     /**
      * @return the dataRows
+     *
+     * @deprecated use getDataRowsAsCompositeVariants
      */
+    @Deprecated
     public List<Map<String, String>> getDataRows() {
+        return dataRows.stream().map(m -> {
+            return m.entrySet().stream().collect(Collectors.toMap(e -> e.getKey(), e ->  e.getValue().toString()));
+        }).collect(Collectors.toList());
```

Comment:
Note that this change isn't strictly backwards compatible. It is *binary* compatible, but the resulting object was originally mutable and is now not guaranteed to be mutable. In general, it would be best to not return mutable objects from methods like this, so I'm not sure if that behavior needs to be guaranteed or not (in which case `Collectors.toCollection(Supplier)` can be used.

Reply:
Sometimes I use supplier because I want specific behavior.  For example LinkedHashSet maintains the order as-inserted and that tends to provide a lot of advantages if you want a map that actually retains a guaranteed order.